pyro/infer/mcmc/hmc.py

```diff
@@ -40,8 +40,7 @@ def _get_trace(self, z):
         for name, value in z.items():
             z_trace.nodes[name]['value'] = value
         trace_poutine = poutine.trace(poutine.replay(self.model, trace=z_trace))
-        trace_poutine(*self._args, **self._kwargs)
-        return trace_poutine.trace
+        return trace_poutine.get_trace(*self._args, **self._kwargs)
```

Comment:
The reason for avoiding the `.get_trace` method is to avoid a trace copy which turns out to be quite expensive (given the number of calls involved) and unnecessary for HMC. Refer to https://github.com/uber/pyro/pull/690#discussion_r163117016 for a discussion on this (you may have to uncollapse the comments).

Reply:
Yeah, I can confirm this! Will change back to the old behavior.